jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/completion/DefaultHandlerCompletion.kt

```diff
@@ -0,0 +1,20 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.completion
+
+import com.intellij.codeInsight.completion.PlainPrefixMatcher
+import com.intellij.codeInsight.completion.PrefixMatcher
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.openapi.project.Project
+import software.aws.toolkits.jetbrains.services.lambda.LambdaHandlerIndex
+
+class DefaultHandlerCompletion : HandlerCompletion {
+
+    override fun getPrefixMatcher(prefix: String): PrefixMatcher =
+        PlainPrefixMatcher(prefix)
+
+    override fun getLookupElements(project: Project): Collection<LookupElement> =
+        LambdaHandlerIndex.listHandlers(project).map { LookupElementBuilder.create(it) }
```

Comment:
FYI the index was removed in master a while back see https://github.com/aws/aws-toolkit-jetbrains/pull/1107

We'll have do to RuntimeGroup specific implementations most likely.

Reply:
Hm. Does that mean you expect no completion in lambda run configuration anymore? Do we expect to support completion in Rider or is it gonna be consistent behavior per all products?

Reply:
We had to remove the index because in 2019.2 we were getting re-entrant indexing errors. Due to the tight deadlines of getting support out it was removed. I am not against bringing back auto-complete, but could it be done with:

1. A better index written by someone knowing what they are doing
2. Just loading completion candidates by filtering the method indexes per runtime group.. (Not sure what performance would look like here...)

We technically never had it working for Python, so we were already a bit inconsistent, but ideally that should be an anti-pattern.

Reply:
@abrooksv, @kiiadi, got it. Thank you.

I could take a look on this issue in Idea and Python. Just a note: I'm using the second approach for Rider since we have all indexes on a backend. I've checked on a solution with ~250 project (+ one SAM Lambda project), the completion popup is showed up almost immediately with a top-level filtering by "lambda-defined" project type.

I'm just wondering what we should do with this commit now :) It makes not a lot of sense after modifications in master. Do you think we should freeze it until master is updated?

Reply:
2nd approach is fine if its fast. The completion system is async anyway so it is not like we block the UI.

I do like this, but maybe it makes sense to get master in and then restore the functionality in this branch. @kiiadi Thoughts?